### ISSUE TITLE
Counts are slightly off on completion for chat models

### ISSUE THREAD

**Issue by arcticfly** [Cainier/gpt-tokens#20]
Awesome project, thank you for adding this to the ecosystem! My brother and I are currently working on https://github.com/openpipe/openpipe, and this package is incredibly useful to us.

I do notice that completion token counts are slightly off on some models. Specifically, it appears that GPTTokens always believes that the completion includes more tokens than it actually does. I created an experiment that compares the number of tokens OpenAI reports were used for a certain response (returned from non-streamed responses) against tokens calculated using GPTTokens (calculated on streamed responses). Here's the experiment: https://openpipe.ai/experiments/e2d5d255-5731-4dbc-9f83-7f642745404d.

I think we're using the latest version (1.0.10): https://github.com/OpenPipe/openpipe/blob/main/package.json#L39

And here are the relevant screenshots:

Non-streamed token counts (read from response):
<img width="368" alt="Screenshot 2023-07-08 at 9 31 26 PM" src="https://github.com/Cainier/gpt-tokens/assets/41524992/6d6f916e-86fb-4412-b4b3-703802068ec0">

Streamed token counts (calculated using GPTTokens):
<img width="414" alt="Screenshot 2023-07-08 at 9 31 15 PM" src="https://github.com/Cainier/gpt-tokens/assets/41524992/d6201c97-1316-494f-a983-111603e08106">

Again, amazing project! Starring now!

**Comment by Yzwcp** [comment]
确实有些误差，不知是什么原因

**Comment by Cainier** [comment]
Thanks for the question, after careful inspection and comparison, I found the problem in the code and will fix it in tomorrow's v1.10.0 release

The cause of the problem (take prompt as an example):

[
   { 'role' : 'user', 'content': 'Hello' },
   { 'role' : 'assistant', 'content': 'Hi, how are you?' },
   { 'role' : 'user', 'content': 'tell a joke' },
     ]

When openai processes Prompt, the entire messages will participate in the tokens calculation, and the same is true for gpt-tokens, so the prompt calculation is correct

But when openai responds to Completion, for example:

{ 'role' : 'assistant', 'content': 'Hi, how are you?' }

openai only calculates tokens for 'content' content, but gpt-tokens calculates tokens for the entire JSON object, so the calculation of tokens is relatively high

---

To solve this problem, some changes will be made to the calling method of the gpt-tokens package in v1.10.0, please refer to the README.md of v1.10.0

**Comment by Cainier** [comment]
> Awesome project, thank you for adding this to the ecosystem! My brother and I are currently working on https://github.com/openpipe/openpipe, and this package is incredibly useful to us.
> 
> I do notice that completion token counts are slightly off on some models. Specifically, it appears that GPTTokens always believes that the completion includes more tokens than it actually does. I created an experiment that compares the number of tokens OpenAI reports were used for a certain response (returned from non-streamed responses) against tokens calculated using GPTTokens (calculated on streamed responses). Here's the experiment: https://openpipe.ai/experiments/e2d5d255-5731-4dbc-9f83-7f642745404d.
> 
> I think we're using the latest version (1.0.10): https://github.com/OpenPipe/openpipe/blob/main/package.json#L39
> 
> And here are the relevant screenshots:
> 
> Non-streamed token counts (read from response): <img alt="Screenshot 2023-07-08 at 9 31 26 PM" width="368" src="https://user-images.githubusercontent.com/41524992/252125905-6d6f916e-86fb-4412-b4b3-703802068ec0.png">
> 
> Streamed token counts (calculated using GPTTokens): <img alt="Screenshot 2023-07-08 at 9 31 15 PM" width="414" src="https://user-images.githubusercontent.com/41524992/252125898-d6201c97-1316-494f-a983-111603e08106.png">
> 
> Again, amazing project! Starring now!

Thanks for the question, after careful inspection and comparison, I found the problem in the code and will fix it in tomorrow's v1.10.0 release

The cause of the problem (take prompt as an example):

``` javascript
const messages = [
  { 'role' : 'user', 'content': 'Hello' },
  { 'role' : 'assistant', 'content': 'Hi, how are you?' },
  { 'role' : 'user', 'content': 'tell a joke' },
]
```

When openai processes Prompt, the entire messages will participate in the tokens calculation, and the same is true for gpt-tokens, so the prompt calculation is correct

But when openai responds to Completion, for example:

``` javascript
console.log(chatCompletion.data.usage)
// { 'role' : 'assistant', 'content': 'this is a joke demo' }
```

openai only calculates tokens for 'content' content, but gpt-tokens calculates tokens for the entire JSON object, so the calculation of tokens is relatively high

---

To solve this problem, some changes will be made to the calling method of the gpt-tokens package in v1.10.0, please refer to the **README.md of v1.10.0**

**Comment by arcticfly** [comment]
Thanks, I appreciate the fast response
